source/client/output_formatter_impl.cc

```diff
@@ -260,6 +263,32 @@ FortioOutputFormatterImpl::durationToSeconds(const Envoy::ProtobufWkt::Duration&
   return Envoy::Protobuf::util::TimeUtil::DurationToNanoseconds(duration) / 1e9;
 }
 
+// Returns true iff the options indicate that Nighthawk was using HTTP/2 or
+// HTTP/3 Quic as the protocol.
+bool isUsingH2OrH3(const nighthawk::client::CommandLineOptions& options) {
+  return options.h2().value() || options.protocol().value() == Protocol::HTTP2 ||
+         options.protocol().value() == Protocol::HTTP3;
+}
+
+// Calculates the number of threads, i.e. the number of connections nighthawk
+// was using when running with the specified number of workers and options.
+uint32_t calculateNumberOfThreads(uint32_t number_of_workers,
+                                  const nighthawk::client::CommandLineOptions& options) {
+  // The stock Envoy h2 pool doesn't offer support for multiple connections
```

Comment:
Thank you for bringing this up, this offers a great simplification.

I am electing to apply it outside of this PR, so that the Quic support can ship. I will send another PR shortly simplifying this code as suggested.

Tracking in https://github.com/envoyproxy/nighthawk/issues/730.